astropy/coordinates/angle_formats.py

```diff
@@ -412,13 +412,29 @@ def degrees_to_dms(d):
 def dms_to_degrees(d, m, s=None):
     """
     Convert degrees, arcminute, arcsecond to a float degrees value.
-    """
+    # solutions to fix the issue #12239 .
+    solution 1 :
+    if d=0 , get the sign from m , if m=0, get sign from s.
 
+    solution 2 :
+    Take sign =1.0 as  another argument.
+    Sign is by default positive and represented by sign = 1.0 .
+    For negative sign, sign = -1.0 , if given by user.
+    Dont use any sign in d,m,s  if sign argument is present.
+    """
     _check_minute_range(m)
     _check_second_range(s)
-
+    # code for  solution 1 :
+    # value of sign by default
+    sign = 1.0;
     # determine sign
-    sign = np.copysign(1.0, d)
+    if (d!=0):
+        sign = np.copysign(1.0, d)
+    # code  for solution 1 :
+    if (((d==0)) and (m!=0)):
+        sign = np.copysign(1.0, m);
+    elif (((d==0) and (m==0)) and (s!=0)):
+        sign = np.copysign(1.0, s)
```

Comment:
The if statement can be simplified by using `elif` to guarantee that previous statements are true.

```suggestion
    if d != 0:
        sign = np.copysign(1.0, d)
    elif m != 0:
        sign = np.copysign(1.0, m);
    elif s != 0:
        sign = np.copysign(1.0, s)
    else:
        sign = 1.0
```

Or even just
```python

    sign = np.copysign(1.0, d if d != 0 else (m if m != 0 else (s if s != 0 else 1.0)))
```